### PR TITLE
feat: Update PSR2, PSR12 and PER-CS2 with `single_space_after_...` config

### PR DESCRIPTION
**Pull request by SpacePossum**
- add config for `single space after` following text and examples of:
    - https://www.php-fig.org/psr/psr-2/
    - https://www.php-fig.org/psr/psr-12/
    - https://www.php-fig.org/per/coding-style/
- add options to force single space before `else` and `elseif` following code-styles (as listed above)
- added short array syntax (PER CS2 @ 11)
- added trailing comma config `array destruction` and unconditionally `parameters` and `match` (following https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8167 and https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8172)

part of on going effort: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7247 and https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/4502

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ A *config* knows about the code style rules and the files and directories that m
...
    * if you work on existing fixers then don't change existing test cases, because these are contract between the maintainers and users (they ensure how tool works). Add new test cases that cover provided changes - preferred way of defining test cases is with [data provider](https://docs.phpunit.de/en/10.0/writing-tests-for-phpunit.html#data-providers) which uses `yield` with proper case description as a key (e.g. `yield 'Some specific scenario' => ['some', 'example', 'data'];`). Codebase may still contain test cases in different format, and it's totally acceptable to use `yield` approach next to existing `return` usages.
-* Update documentation: `composer docs`. This requires the highest version of PHP supported by PHP CS Fixer. If it is not installed on your system, you can run it in a Docker container instead: `docker compose run php-8.2 php dev-tools/doc.php`.
+* Update documentation: `composer docs`. This requires the highest version of PHP supported by PHP CS Fixer. If it is not installed on your system, you can run it in a Docker container instead: `docker-compose run php-8.2 php dev-tools/doc.php`.
```
Comment (kubawerlos) [inline CONTRIBUTING.md:34]: I'm not about this one, I saw @Wirone did the opposite: https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7450/commits/88f703fd7f750b327f3929c91ec4fed4f53d6b8a

Comment (SpacePossum) [reply CONTRIBUTING.md:34]: :+1:  lets see what Wirone thinks,
I get:
```
$ docker compose
docker: 'compose' is not a docker command.
```

**Comment by SpacePossum** [comment]
rebased on latest and forced without other changes

**Comment by SpacePossum** [comment]
I don't understand the mac os fail

**Comment by SpacePossum** [comment]
rebase on latest of main,
added commit https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8171/commits/f8bc3db321b9d3c5d6136bc26c8b38351b706822
no other changes

**Comment by SpacePossum** [comment]
rebase on latest of main, no other changes

**Review by kubawerlos — APPROVED** [review]

**Comment by kubawerlos** [comment]
Thanks @SpacePossum

**Comment by SpacePossum** [comment]
Thanks!

**Comment by keradus** [comment]
fantastic!

**Comment by Wirone** [comment]
@SpacePossum could you help me with finding points that should be checked out in 7247 and 4502? I find it hard to review this huge change and correlate it with todo lists 😅.

**Comment by SpacePossum** [comment]
please for details about what I found about 7247 and 4502 see:
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7247#issuecomment-2294813015
https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/4502#issuecomment-2308178312

**Pull request merged**